modin/pandas/base.py

```diff
@@ -2213,6 +2214,33 @@ def rank(
             )
         )
 
+    def _copy_index_metadata(self, source, destination):  # noqa: PR01, RT01, D200
+        """
+        Copy Index metadata from `source` to `destination` inplace.
+        """
+        if hasattr(source, "name") and hasattr(destination, "name"):
+            destination.name = source.name
+        if hasattr(source, "names") and hasattr(destination, "names"):
+            destination.names = source.names
+        return destination
+
+    def _ensure_index(self, index_like, axis=0):  # noqa: PR01, RT01, D200
+        """
+        Ensure that we have an index from some index-like object.
+        """
+        if (
+            self._query_compiler.has_multiindex(axis=axis)
+            and is_list_like(index_like)
```

Comment:
otherwise, it'll make us to copy a MultiIndex `index_like` object
```suggestion
            and not isinstance(index_like, pandas.MultiIndex)
            and is_list_like(index_like)
```

Reply:
Here it is better to use `Index` type.